web/src/js/components/Widget/Widgets/Search/Search.js

```diff
@@ -41,7 +62,19 @@ class Search extends React.Component {
     const engine = this.state.config.engine || 'Google'
```

Comment:
Here, if isYahooUser, set it to always be Yahoo. Then, we don't have to worry about the widget settings at all.